src/pages/level/Controls.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  Agriculture as AgricultureIcon,
+  BusAlert as BusIcon,
+  Call as CallIcon,
+  Cast as CastIcon,
+  ChevronLeft as ChevronLeftIcon,
+  Menu as MenuIcon,
+} from "@mui/icons-material"
+import {
+  type CSSObject,
+  Divider,
+  Drawer,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  type Theme,
+} from "@mui/material"
+import { type FC, type ReactNode, useState } from "react"
+
+export interface MiniDrawerItem {
+  icon: ReactNode
+  text: string
+}
+
+export interface ControlsProps {}
+
+export interface MiniDrawerListProps {
+  isDrawerOpen: boolean
+  items: MiniDrawerItem[]
+}
+
+export interface MiniDrawerProps {
+  open: boolean
+  children: ReactNode
+  onToggle: () => void
+}
+
+const DRAWER_WIDTH = 240
+
+const openedMixin = (theme: Theme): CSSObject => ({
+  width: DRAWER_WIDTH,
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  }),
+})
+
+const closedMixin = (theme: Theme): CSSObject => ({
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  width: `calc(${theme.spacing(7)} + 1px)`,
+  [theme.breakpoints.up("sm")]: {
+    width: `calc(${theme.spacing(8)} + 1px)`,
+  },
+})
+
+const MiniDrawer: FC<MiniDrawerProps> = ({ open, onToggle, children }) => (
```

Comment:
inline props since component is not exported
```suggestion
const MiniDrawer: FC<{
  open: boolean
  children: ReactNode
  onToggle: () => void
}> = ({ open, onToggle, children }) => (
```